packages/babel-parser/src/parser/expression.js

```diff
@@ -380,6 +380,58 @@ export default class ExpressionParser extends LValParser {
 
         node.right = this.parseExprOpRightExpr(op, prec, noIn);
 
+        /* this check is needed because for operations like
+         * a ?? b || c
+         * a ?? b => This is considered as a logical expression in the ast tree
+         * c => Identifier
+         * so we need to check, only for || cases that if left is logical expression without paren
+         * then raise an error
+         */
+
+        if (op === tt.logicalOR) {
```

Comment:
We can actually avoid this by lowering the precedence of the `??` operator compared to `||`. You can do that by raising the `binop` for the operators in https://github.com/babel/babel/blob/4d30379d3657142cb079a41f37642f519d699593/packages/babel-parser/src/tokenizer/types.js#L142-L156. (This is actually how the spec is implemented, `??` has lower precedence than `||` and the rest)

Reply:
Thank you @jridgewell. This change has been made.